validator/attest.go

```diff
@@ -41,24 +41,59 @@ func Attest(config *Config, logger utils.ZapLogger) error {
 	defer wg.Wait()
 	defer close(dispatcher.AttestRequired)
 
-	// Subscribe to the block headers
-	wsProvider, headersFeed, err := BlockHeaderSubscription(config.Provider.Ws, &logger)
-	if err != nil {
-		return err
-	}
-	defer close(headersFeed)
-	defer wsProvider.Close()
-
-	if err := ProcessBlockHeaders(headersFeed, signer, &logger, &dispatcher); err != nil {
-		return err
-	}
 	// I'd also like to check the balance of the address from time to time to verify
 	// that they have enough money for the next 10 attestations (value modifiable by user)
 	// Once it goes below it, the console should start giving warnings
 	// This the least prio but we should implement nonetheless
-
+	//
 	// Should also track re-org and check if the re-org means we have to attest again or not
-	return nil
+
+	return RunBlockHeaderWatcher(config, &logger, signer, &dispatcher, wg)
+}
+
+func RunBlockHeaderWatcher[Account Accounter, Logger utils.Logger](
+	config *Config,
+	logger Logger,
+	signer Account,
+	dispatcher *EventDispatcher[Account, Logger],
+	wg *conc.WaitGroup,
+) error {
+	// TODO: Should we give a default retry here ? by reusing DEFAULT_MAX_RETRIES (10) for example
```

Comment:
I think it's ok to keep going up until eternity if the error is external. There is no correct choice, after having an idea of how people expect the tool to behave, I think it's what we should do.

Just run until the external service is back again